fiboa_cli/publish.py

```diff
@@ -0,0 +1,378 @@
+            "provider": self.converter.provider,
+            "license": self.converter.license,
+            "projection": "",
+            "homepage": "",
+            "submitter": "Fiboa project",
+        }
+        properties = {}
+        if text:
+            data.update({a.lower(): b for a, b in re.findall(r"- \*\*(.+?):\*\* (.+?)\n", text)})
+            properties = {
+                mapping.get(a.lower(), a.lower()): b.strip()
+                for a, b in re.findall(r"\n\|\s*(\w+)[^|]*\|[^|]*\|[^|]*\|([^|]*)\|", text)
+            }
+
+        return data, properties
+
+    def readme_attribute_table(self, stac_data, properties):
+        def description(name):
+            m = self.converter.columns
+            reverse = dict(zip(m.values(), m.keys()))
+            return (
+                properties.get(reverse.get(name))
+                or properties.get(name)
+                or DESCRIPTIONS.get(name, "")
+            )
+
+        cols = [["Property", "**Data Type**", "Description"]] + [
+            [
+                s["name"],
+                re.search(r"\w+", s["type"])[0],
+                description(s["name"]),
+            ]
+            for s in stac_data["assets"]["data"]["table:columns"]
+            if s["name"] not in ("geometry", "bbox", "collection")
+        ]
+        widths = [max(len(c[i]) for c in cols) for i in range(3)]
+        aligned_cols = [[f" {c:<{w}} " for c, w in zip(row, widths)] for row in cols]
+        aligned_cols.insert(1, ["-" * (w + 2) for w in widths])
+        return "\n".join(["|" + "|".join(cols) + "|" for cols in aligned_cols])
+
+    def make_license(self):
+        text = ""
+        try:
+            data, properties = self.get_data_survey()
+            text = data["license"]
+            if getattr(self.converter, "license") not in (None, "", data["license"]):
+                text += "\n" + self.converter.license + "\n"
+
+            found = False
+            for _license in (data["license"], self.converter.license):
+                if not _license or "<(https://" in _license:
+                    continue
+
+                # Include full-license text
+                _license = _license.upper()
+                if _license in spdx_license_list.LICENSES:
+                    response = requests.get(
+                        f"https://raw.githubusercontent.com/spdx/license-list-data/refs/heads/main/text/{_license}.txt"
+                    )
+                    if response.ok:
+                        found = True
+                        text += f"\n\n{response.text}\n"
+                        break
+            if not found:
+                self.warning(f"License {text} could not be found in SPDX license list")
+
+        except Exception as e:
+            self.exception(e)
+        return text
+
+    def make_readme(self, file_name, stac):
+        version = Registry.get_version()
+        converter = self.converter
+        stac_data = json.load(open(stac))
```

Comment:
Is this closing the resources properly? I always see these with statements in Python, not sure whether this is also valid.

Reply:
Well, if you trust/rely on a reference-counted garbage collector it's same same, but I'll change it...